docs/middleware.rst

```diff
@@ -198,9 +213,11 @@ To add or remove items in different layers, use the following API:
     .. code-block:: python
 
         >>> w3 = Web3(...)
-        >>> w3.middleware_onion.add(web3.middleware.pythonic_middleware)
+        >>> w3.middleware_onion.add(web3.middleware.gas_price_strategy_middleware)
+        # or
+        >>> w3.middleware_onion.add(web3.middleware.gas_price_strategy_middleware, 'gas_price_strategy')
         # or
-        >>> w3.middleware_onion.add(web3.middleware.pythonic_middleware, 'pythonic')
+        >>> w3.middleware_onion.add(web3.middleware.async_gas_price_strategy_middleware, 'gas_price_strategy')
```

Comment:
async middleware should only go on an async provider. Adding the following above the async examples would work:

```py
async_w3 = AsyncWeb3(...)
```

Reply:
Good catch, updated!